R/learner_blockForest_surv_blockforest.R

```diff
@@ -0,0 +1,122 @@
+#'
+#' @templateVar id surv.blockforest
+#' @template learner
+#'
+#' @references
+#' `r format_bib("hornung2019blockforest")`
+#'
+#' @template seealso_learner
+#' @examplesIf mlr3misc::require_namespaces(c("mlr3proba"), quietly = TRUE)
+#' # Define a Task
+#' task = tsk("grace")
+#' # Create train and test set
+#' ids = partition(task)
+#' # check task's features
+#' task$feature_names
+#' # partition features to 2 blocks
+#' blocks = list(bl1 = 1:3, bl2 = 4:6)
+#' # define learner
+#' learner = lrn("surv.blockforest", blocks = blocks,
+#'               importance = "permutation", nsets = 10,
+#'               num.trees = 50, num.trees.pre = 10, splitrule = "logrank")
+#' # Train the learner on the training ids
+#' learner$train(task, row_ids = ids$train)
+#' # feature importance
+#' learner$importance()
+#' # Make predictions for the test observations
+#' pred = learner$predict(task, row_ids = ids$test)
+#' pred
+#' # Score the predictions
+#' pred$score()
+#' @export
+LearnerSurvBlockForest = R6::R6Class("LearnerSurvBlockForest",
+  inherit = mlr3proba::LearnerSurv,
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function() {
+      param_set = paradox::ps(
+        blocks              = p_uty(tags = c("train", "required")),
+        block.method        = p_fct(c("BlockForest", "RandomBlock", "BlockVarSel", "VarProb", "SplitWeights"), default = "BlockForest", tags = "train"), #nolint
+        num.trees           = p_int(1L, default = 2000L, tags = "train"),
+        mtry                = p_uty(default = NULL, tags = "train"),
+        nsets               = p_int(1L, default = 300L, tags = "train"),
+        num.trees.pre       = p_int(1L, default = 1500L, tags = "train"),
+        splitrule           = p_fct(c("logrank", "extratrees", "C", "maxstat"), default = "extratrees", tags = "train"),
+        always.select.block = p_int(0L, 1L, default = 0L, tags = "train"),
+        # adding few key parameters from blockForest::blockForest()
+        importance          = p_fct(c("none", "impurity", "impurity_corrected", "permutation"), tags = "train"),
+        num.threads         = p_int(1L, default = 1L, tags = c("train", "predict", "threads")),
+        seed                = p_int(default = NULL, special_vals = list(NULL), tags = c("train", "predict")),
+        verbose             = p_lgl(default = TRUE, tags = c("train", "predict"))
+      )
+
+      param_set$values = list(num.threads = 1)
+
+      super$initialize(
+        id = "surv.blockforest",
+        param_set = param_set,
+        predict_types = c("crank", "distr"),
+        feature_types = c("logical", "integer", "numeric", "factor", "ordered"),
+        properties = c("weights", "importance"),
+        packages = c("mlr3extralearners", "blockForest"),
+        label = "Block Forests: Random Forests for Blocks of Clinical and Omics Covariate Data"
```

Comment:
Maybe a shorter label

Reply:
done! changed to "Block Forests: Random Forests for Block-wise Data"